### DISCUSSION TITLE
What's new ?

### DISCUSSION THREAD

**Discussion by mak-gitdev** [mak-gitdev/HA_enoceanmqtt#17] [Announcements]
This thread gives update on the work ([as requested :blush:](https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/14#discussioncomment-4219110))

⚠️
**If you have a custom mapping.yaml file, before upgrading to a new version, leave empty the configuration entry `mapping_file` or merge your modifications into the [latest version](https://github.com/mak-gitdev/HA_enoceanmqtt/blob/develop/enoceanmqtt/overlays/homeassistant/mapping.yaml), so that the latest added features are available.**
⚠️

⚠️
**If you use a device that works well but is marked as not tested in the [list of supported devices](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Supported-devices), please report it in the comments so I can update the status of this device.
Thanks 👍 !**
⚠️

## To Do List
[To Do List](https://github.com/users/mak-gitdev/projects/1)

### 2024/02/01
From https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/57#discussioncomment-8340082:
<blockquote>
Hi everyone,

There is a [preview version](https://github.com/mak-gitdev/hidden-addon) available for the support of Eltako devices.

Supported devices are listed [here](https://github.com/mak-gitdev/hidden-addon/blob/master/addon-test-eltako/enoceanmqtt.devices.sample) and you can have a look at this [discussion](https://github.com/mak-gitdev/hidden-addon/discussions/2) to set up your devices.
If your device is not listed there, note that actuators are controlled by sending GFVS telegrams, so look into the user manual of your device to know how to configure it to work with GFVS.
If you manage to set up a device not listed in that discussion, please add the instructions there so others can benefit from your experience 🙂. 

This version will later be merged in HA_enoceanmqtt 0.2.1.
In the meantime, you can use it and let me know if you have any issues.
To report an issue with this preview version or discuss about this version, **please use the [discussion panel](https://github.com/mak-gitdev/hidden-addon/discussions) of the dedicated repository**.
</blockquote>

### 2023/11/28
Release dev version [0.1.28](https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/102)

### 2023/10/21
Release dev version [0.1.27](https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/89)

### 2023/08/10
**You can now take a closer look at the work in progress on this [page](https://github.com/users/mak-gitdev/projects/1)**.

### 2023/06/05
Release dev version 0.1.26:
- Latest enocean-mqtt version which fixes status bits setting (see mak-gitdev/HA_enoceanmqtt#65 and embyt/enocean-mqtt#41 for more details)
- From now, use mak-gitdev/enocean as Python EnOcean Library which fixes "list index out of range error" thanks to @kridgo patch (see kipe/enocean#134 and kipe/enocean#138 for more details).
- Add virtual A5-10-06 (mak-gitdev/HA_enoceanmqtt#46)
- Add '%' as unit of measurement for A5-20-01 current value field (mak-gitdev/HA_enoceanmqtt#53)
- Update F6-02-[01-02] mapping to send status bits thanks to @LarsKoeppel.

### 2023/02/20
Release dev version 0.1.25:
- Fix issue #41 

### 2023/02/17
Release dev version 0.1.24:
**If you have a custom mapping.yaml file, before starting the new version, leave empty the configuration entry `mapping_file` or merge your modifications into the [latest version](https://github.com/mak-gitdev/HA_enoceanmqtt/blob/develop/enoceanmqtt/overlays/homeassistant/mapping.yaml), so that the latest added features are available.**
- **Change devices and entities unique ids. This may lead to some devices loosing some configurations such as zones, etc.**
- Add better support for A5-13-01. ID1, ID2 and ID3 are now supported. ID4 to ID6 will follow in another release.
- Use device triggers for D2-03-0A instead of binary sensors.
- Add support for virtual devices. **We can now declare more than one device sending broadcast data**. Only F6-02-01 and F6-02-02 are available for the moment. Have a look at this [wiki chapter](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Usage#virtual-devices) for more details.
- Add EEP documentation for each device in device page.
![image](https://user-images.githubusercontent.com/73297795/219525657-64322fe9-0a87-4a98-9b1d-3bb44fc29bd3.png)
- Add packet receipt date to all devices so that it is possible to know the last time the device has been seen.
![image](https://user-images.githubusercontent.com/73297795/219525965-5cdfc17f-775d-42cc-acb8-bc7cd900c747.png)
- Add support for:
  - D2-01-0A
  - D2-01-0D
  - D2-01-0E
  - A5-13-[01-06]
  - A5-10-03
  - A5-10-05
  - A5-10-06
  - A5-10-10
  - A5-10-12
  - A5-08-[01-03]
  - A5-14-01
  - A5-14-0A
  - A5-20-01 (no control yet, only status)
  - A5-20-04 (no control yet, only status)
  - Initial support for A5-38-08 command 1
- Add measurement control entities for D2-01-0B, D2-01-0C and D2-01-0E.
![image](https://user-images.githubusercontent.com/73297795/219526968-4d265e66-8642-45a8-89a0-b0d8a96cc9cc.png)
- Add a new configuration entry `eep_file` to add support for new EEPs at Python EnOcean library level.
- Remove some unused entities from D2-01-xx devices to comply with the EEP documentation.
- Remove delete entity from devices. Devices can now be deleted using the integrated MQTT delete button.
![image](https://user-images.githubusercontent.com/73297795/219527397-9e9d811d-e931-4295-816f-be6c9666f134.png)
- Fix issue mak-gitdev/HA_enoceanmqtt#37

### 2023/01/10

A docker image is now available so that you can now use HA_enoceanmqtt directly in Home Assistant as an addon, or standalone as an install on a Linux system or as a docker container (typically for those using Home Assistant container).

Depending on your architecture, you can now use:
```
docker pull makgitdev/ha_enoceanmqtt_dev-amd64
```
or
```
docker pull makgitdev/ha_enoceanmqtt_dev-aarch64
```
or
```
docker pull makgitdev/ha_enoceanmqtt_dev-armhf
```

Then:
- Mount the `/config` volume and your enocean device.
- Adapt the [enoceanmqtt.conf.sample](https://github.com/mak-gitdev/HA_enoceanmqtt/blob/14f03549537562a1f1f067f6844ce4b2947ddd31/standalone/enoceanmqtt.conf.sample) file in the `/config` folder to set parameters as well as devices to manage.

The volume mount has to point to -v /local/path/to/configfile:/config.
Or pass a docker volume created using the `docker volume create` command.

Example docker command:
```
sudo docker run --device=/dev/enocean -v my_docker_volume:/config makgitdev/ha_enoceanmqtt_dev-aarch64
```
You can add `--debug` to enable debug messages.

### 2022/12/31
- **Move addon to a new repository. See [HA_enoceanmqtt-addon](https://github.com/mak-gitdev/HA_enoceanmqtt-addon) to see [how to migrate to the new addon repository](https://github.com/mak-gitdev/HA_enoceanmqtt-addon#migration-from-the-old-addon-repository).**
- Release dev version 0.1.23:
  - Rename not rounded entities `xx_temperature` and `xx_humidity` from A5-04-XX devices to `t_raw` and `h_raw`. These new entities are also disable by default.
  - Add support for:
    - A5-02-[01-...-0B]
    - A5-02-[10-...-1B]
    - A5-02-20
    - A5-02-30
    - A5-06-[01-...-02]
    - A5-07-[01-...-03]
    - A5-13-01 (ID 1)
    - D2-03-0A
  - Remove addon from repository
  - Remove °F entities from A5-04-[01-...-04] devices as HA seems to convert automatically °C to °F and vice-versa
  - Fix status entity for A5-04-[01-...-04]  devices.

### 2022/11/25
- Creation of develop branch to test bugfix and new features.

- Creation of a new addon tracking the development branch. You may want to use this one for latest features and bugfixes.
![image](https://user-images.githubusercontent.com/73297795/204041478-879ea44e-4b82-486f-8b53-a554814437e5.png)

- Latest script version may fix issue #10 (waiting for confirmation).

- Add support for A5-04-[01-...-04] devices as requested in #12 and #4 

- EnOcean Base ID now reported in the MQTT integration on ENOCEANMQTT device
![image](https://user-images.githubusercontent.com/73297795/204040628-564485ed-d363-492f-90bf-f1e6be9a54b4.png)

**Comment by commentsonly** [comment]
installed smoothly 👍 
haven't updated the mappings yet since I wanted to gain some feedback from you abou an idea ... I'll start a new topic
